src/io.jl

```diff
@@ -9,18 +11,31 @@ function read_csv(path, sink::Type{A}; kws...) where A <: AbstractMatrix
     A(read_csv(path; kws...))
 end
 
-function read_csv(path, sink::Type{<:DataFrame}; kws...)
-    return CSV.read(path, sink; kws...)
+function read_csv_asdf(path; kws...)
+    DF = checked_import(idDataFrames).mod.DataFrame
+    return checked_import(idCSV).read(path, DF; kws...)
 end
 
 function read_npy(path)
-    return Pickle.npyload(path)
+    return FileIO.load(File{format"NPY"}(path))
 end
 
 function read_npz(path)
-    return NPZ.npzread(path)
+    return FileIO.load(File{format"NPZ"}(path))
 end
 
 function read_pytorch(path)
-    return Pickle.Torch.THload(path)
+    Base.invokelatest(checked_import(idPickle).mod.Torch.THload, path)
+end
+
+function read_pickle(path)
+    checked_import(idPickle).npyload(path)
```

Comment:
The world age issue is because many method definitions are not yet "refreshed" when you do dynamic lazy call from an old world.

This is exactly where `invokelatest` is used: it calls the method from the latest world where the methods are refreshed.

`invokelatest` has some small overhead, so if the dependencies are not that big, we can preload them via, e.g., `using Pickle` as we used to do.
But yes, loading a predefined __small__ dataset is definitely a perfect case where `invokelatest` overhead is acceptable.

```suggestion
    Base.invokelatest(checked_import(idPickle).npyload, path)
```

Similar changes should be made elsewhere.